packages/postgrest/lib/src/postgrest_transform_builder.dart

```diff
@@ -182,8 +182,14 @@ class PostgrestTransformBuilder<T> extends PostgrestBuilder<T, T> {
   ///
   /// By specifying this type via `.select<Map<String,dynamic>?>()` you get more type safety.
   PostgrestTransformBuilder<T> maybeSingle() {
-    _headers['Accept'] = 'application/vnd.pgrst.object+json';
-    _maybeEmpty = true;
+    // Temporary fix for https://github.com/supabase/supabase-flutter/issues/560
+    // Issue persists e.g. for `.insert([...]).select().maybeSingle()`
+    if (_method?.toUpperCase() == 'GET') {
+      _headers['Accept'] = 'application/json';
```

Comment:
What about adding a `limit=2` here as well? This would increase the performance, becauset the payload doesn't get that big.

Reply:
Hmm, I would rather not do anything that the user didn't ask for. I also fear that down the line there could be a case where we shoot ourselves on the foot.

Reply:
Using `maybeSingle` already requires developers to make sure that it only returns 0 or 1 results anyway, so it shouldn't matter. Anyway, I have edited the comment doc of the `maybeSingle` method slightly.